tests/test_global.py

```diff
@@ -0,0 +1,292 @@
+    ), "Certain columns are not in the links"
+
+
+def test_pneo_links():
+    # Test if PNEO links are found
+    pneo_links = links[links["resolution"] == "30cm"]
+    pneo_links = pneo_links[pneo_links["link"].str.contains("RGBNED")]
+    pneo_links = pneo_links.sort_values("percentage_geojson")
+
+    assert len(pneo_links) > 0, "No PNEO links found error!"
+
+
+def test_superview_links():
+    # Example filter on resolution and bands
+    superview_links = links[links["resolution"] == "50cm"]
+    superview_links = superview_links[superview_links["link"].str.contains("RGBI")]
+    superview_links = superview_links.sort_values("percentage_geojson")
+
+    assert len(superview_links) > 0, "No  PNEO links found error!"
+
+
+# Cropping tests
+
+
+def pneo_polygon_crop():
+
+    georegion = nso.nso_georegion(
+        path_to_geojson="C:/repos/satellite-images-nso/input_data/Test_region.geojson",
+        output_folder=output_path,
+        username=nso_username,
+        password=nso_password,
+    )
+
+    filepath = georegion.execute_link(
+        "https://api.satellietdataportaal.nl/v1/download/30cm_RGBNED_12bit_PNEO/20230513_104139_PNEO-03_1_1",
+        plot=False,
+    )
+
+    assert filepath, "No file has been downloaded or extracted for a polygon in PNEO!"
+
+    # Remove the file for later tests
+    os.remove(filepath)
+
+
+def pneo_multiploygon_crop():
+
+    georegion = nso.nso_georegion(
+        path_to_geojson="C:/repos/satellite-images-nso/input_data/Test_multipolygon_region.geojson",
+        output_folder=output_path,
+        username=nso_username,
+        password=nso_password,
+    )
+
+    filepath = georegion.execute_link(
+        "https://api.satellietdataportaal.nl/v1/download/30cm_RGBNED_12bit_PNEO/20230513_104139_PNEO-03_1_1"
+    )
+
+    assert filepath, "No file has been downloaded or extracted for multiPolygons"
+
+    # Remove the file for later tests
+    os.remove(filepath)
+
+
+# Channel adding tests.
+
+
+def test_re_ndvi_pneo():
+    # Test re_ndvi on PNEO coepelduynen and see if gives the best results.
+    path_geojson = "C:/repos/satellite-images-nso/input_data/Test_region.geojson"
+    output_path = "E:/data/test"
+
+    georegion = nso.nso_georegion(
+        path_to_geojson=path_geojson,
+        output_folder=output_path,
+        username=nso_username,
+        password=nso_password,
+    )
+
+    filepath = georegion.execute_link(
+        "https://api.satellietdataportaal.nl/v1/download/30cm_RGBNED_12bit_PNEO/20230513_104139_PNEO-03_1_1",
+        add_red_edge_ndvi_band=True,
+        plot=False,
+    )
+
+    # Open the TIFF image and display ndvi
+    with rasterio.open(filepath) as src:
+        # Get the number of bands
+        num_bands = src.count
+        print("Number of bands:", num_bands)
+
+        assert num_bands >= 7, "re_ndvi bands not added correctly"
+
+
+def test_adding_ndwi_pneo():
+    # Add the NDWI Channel
+    filepath_ndwi = nso_manipulator.add_index_channels(
```

Comment:
What would happen to this test if the previous test fails? You should try to make tests that are completely independent. Then when a test fails it is immediate apparent where the problem lies.

Reply:
should be fixed